### PR TITLE
Capstone Ruby on Rails: Smartest 

### PR DESCRIPTION
**Pull request by marurevi**
### Design
I followed the [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) of [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).
### Interactions
As you can see on the video, the page has the following pages and features:

#### Splash screen

> A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
> Sign up and log in pages

> The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
> If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

#### Home page (catagories page)

> When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
> When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
> There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

#### Transactions page

> For a given category, the list of transactions is presented, ordered by the most recent.
> At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
> There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
> When the user clicks on the "Back" button (<), the user navigates to the home page.

#### "Add a new category" page

> The user fills out a form to create a new category, indicating their name and icon (both mandatory).
> The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
> When the user clicks on the "Back" button (<), the user navigates to the home page.

#### "Add a new transaction" page

> The user fills out a form to create a new transaction with:
  - name (mandatory)
  - amount (mandatory)
  - categories (mandatory at least one)

> The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
> When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
I created unit and integration tests for all the most important components of your RoR application.
I get the 97% of coverage of the full application like you can see in the picture below.

image

#### Mi loom video:

https://www.loom.com/share/c75d70a001a14d149a56e250bc0287e1

### CODE REVIEW

**Comment by marurevi** [comment]
I had some issues deploying the app. I'm working with WSL (Windows Subsystem for Linux) and it seems incompatible with Heroku. In a previous module, I deployed in Heroku from windows. However, I haven't installed Postgres on windows and I'm afraid that it take me more time that I expect and don't be able to achieve the deadline.
Regards
**_Mariana Revilla_**

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @marurevi , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md [hunk 1]
```diff
@@ -1,52 +1,94 @@
...
+- Ruby on Rails
+- Boostrap
+- Postgres SQL
+- Rspec
+- Capybara
 
 ## Live Demo (if available)
 
-[Live Demo Link](https://livedemo.com)
+[Live Demo Link](https://budget-app-smartes.herokuapp.com/)
```
[the remark this answers is not in the document]
Comment (marurevi) [reply README.md:21]: @AnselemOdims Indeed, I had technical issues as I described above. I will ask for help from a TLA but I'm afraid of not getting that solved on time. However, you can clone the repository and you will be able to check the app. It works...

**Review by rotimiazeez — APPROVED** [review]
# Hi @marurevi 👏🏽, 
## You did great on this project 👍🏽
### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**